File: C++/0009_palindromenumber.cc
// 2020-08-04 23:14:52

class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;
        int remain = x;
        long sum = 0;
        while (remain != 0) {
            int num = remain % 10;
            sum = sum * 10 + num;
            remain /= 10;
        }
        return sum == x;
    }
};

File: TypeScript/0009-palindrome-number.ts
// 2020-08-05 15:26:18

function isPalindrome(x: number): boolean {
  if (x < 0) {
    return false;
  }
  let remain = x;
  let sum = 0;
  while (remain != 0) {
    let num = remain % 10;
    sum = sum * 10 + num;
    remain = Math.floor(remain / 10);
  }
  return sum == x;
}
